rllib/env/multi_agent_episode.py

```diff
@@ -2325,6 +2293,8 @@ def _get_data_by_env_steps_as_list(
                     hanging_val,
                     filter_for_skip_indices=idxes[i],
                 )
+                if what == "extra_model_outputs" and not inf_lookback_buffer:
```

Comment:
I guess `inf_lookback_buffer` is a bool. Imo the code becomes better readable, if we change the return values of `_get_inf_lookback_buffer_or_dict` to a `indices_to_use` only and instead check here for the type.